### ISSUE TITLE
time.Duration should implement encoding.TextMarshaler

### ISSUE THREAD

**Issue by nicksnyder** [golang/go#24174]
time.Time implements all of the marshaler and unmarshaler interfaces, but time.Duration implements none of these.

I came across a case today where it would have been very convenient if time.Duration had already implemented encoding.TextMarshaler and encoding.TextUnmarshaler.

Would you accept a contribution that added implementations for these interfaces?

**Comment by bontibon** [comment]
> Would you accept a contribution that added implementations for these interfaces?
Not until Go2.

Duplicate of #10275.

**Comment by dsnet** [comment]
Closing as duplicate.
